internal/person/person_test.go

```diff
@@ -0,0 +1,171 @@
+package person
+
+import "testing"
+
+// func (p Person) Greet() string {
+// 	return fmt.Sprintf("Hello %s", p.Name)
+// }
+
+func TestPersonGreet(t *testing.T) {
+	anna := Person{
+		Name:   "Anna",
+		Age:    17,
+		Height: 160,
+		Weight: 49,
+	}
+	test := anna.Greet()
+	wantedResult := "Hello Anna"
+	if test != wantedResult {
+
+		t.Errorf("anna.Greet() = %v; want %v", test, wantedResult)
+	}
+}
+
+func TestBMI(t *testing.T) {
+	anna := Person{
+		Name:   "Anna",
+		Age:    17,
+		Height: 160,
+		Weight: 49,
+	}
+	test := anna.BMI()
+	wantedResult := 19.140625
+	if test != wantedResult {
+
+		t.Errorf("anna.BMI() = %v; want %v", test, wantedResult)
+	}
+}
+
+func TestIsAdult(t *testing.T) {
+	anna := Person{
```

Comment:
Make more than 1 test: test on < 18, = 18, > 18

Reply:
Added

Reply:
You should switch the test to use an array of struct:

```go
tests := []struct {
  p Person
  isAdult bool
} {
  // ...
}
```

Reply:
Done.